client.go

```diff
@@ -74,6 +156,141 @@ func SelectOneOf(protos []string, rwc io.ReadWriteCloser) (string, error) {
 	return "", ErrNotSupported
 }
 
+func simOpen(protos []string, rwc io.ReadWriteCloser) (string, bool, error) {
+	mynonce := make([]byte, 32)
+	_, err := rand.Read(mynonce)
+	if err != nil {
+		return "", false, err
+	}
+
+	werrCh := make(chan error, 1)
+	go func() {
+		myselect := []byte(tieBreakerPrefix + base64.StdEncoding.EncodeToString(mynonce))
+		err := delimWriteBuffered(rwc, myselect)
+		werrCh <- err
+	}()
+
+	// skip exactly one protocol
+	// see https://github.com/multiformats/go-multistream/pull/42#discussion_r558757135
+	_, err = ReadNextToken(rwc)
+	if err != nil {
+		return "", false, err
+	}
+
+	// read the tie breaker nonce
+	tok, err := ReadNextToken(rwc)
+	if err != nil {
+		return "", false, err
+	}
+	if !strings.HasPrefix(tok, tieBreakerPrefix) {
+		return "", false, errors.New("tie breaker nonce not sent with the correct prefix")
+	}
+
+	if err = <-werrCh; err != nil {
+		return "", false, err
+	}
+
+	peernonce, err := base64.StdEncoding.DecodeString(tok[7:])
+	if err != nil {
+		return "", false, err
+	}
+
+	var iamserver bool
+	switch bytes.Compare(mynonce, peernonce) {
+	case -1:
+		// peer nonce bigger, he is client
+		iamserver = true
+
+	case 1:
```

Comment:
You actually don't need this case, `iamserver` is already initialized to `false`.

Reply:
I think this helps readability of the code a bit, so keeping it around.